test/parallel/test-loader.js

```diff
@@ -0,0 +1,56 @@
+import { tryPackage }  from '../../lib/internal/modules/cjs/loader'
```

Comment:
`test-loader.js` is a much too generic name. `test-require-package-errors.js` perhaps?

Or even better, find where we currently test the errors potentially thrown by `require` and add your new cases there.